rfc/rfc-44/rfc-44.md

```diff
@@ -0,0 +1,156 @@
+- **ConnectorFactory**: the factory class to create connector instances. `HudiConnectorFactory` which implements
+  ConnectorFactory is to be added for Hudi connector.
+- **Connector**: the facade class of all service classes for a connector. `HudiConnector` which implements Connector
+  is to be added for Hudi connector. Primary service classes for Hudi connector are listed below.
+    - **ConnectorMetadata**: the service class to retrieve (and update if possible) metadata from/to a data source. 
+      `HudiMetadata` which implements ConnectorMetadata is to be added to Hudi connector.
+    - **ConnectorSplitManager**: the service class to generate ConnectorSplit for accessing the data source. 
+      A ConnectorSplit is similar to a Split in Hadoop MapReduce computation paradigm. `HudiConnectorSplitManager`
+      which implements ConnectorSplitManager, is to be added to Hudi connector.
+    - **ConnectorPageSourceProvider**: the service class to generate a reader to load data from the data source.
+      `HudiPageSourceProvider` which implements ConnectorPageSourceProvider, is to be added to Hudi connector. 
+      
+There are other service classes (e.g. `ConnectorPageSinkProvider`), which are not covered in this RFC but might be
+implemented in the future. A class-diagrammatic view of the different components is shown below.
+
+![](presto-connector.png)
+
+### HudiMetadata
+
+HudiMetadata implements ConnectorMetadata interface, and provides methods to access the metadata like databases
+(called schemas in Presto), table and column structures, statistics and properties, and other info.
+As Hudi table metadata is synchronized to Hive MetaStore, HudiMetadata reuses ExtendHiveMetastore from Presto codebase
+to retrieve Hudi table metadata. Besides, HudiMetadata takes use of HoodieTableMetaClient to expose Hudi specified
+metadata, e.g. instants list of a table timeline, represented as SystemTable.
+      
+### HudiSplitManager & HudiSplit
+
+HudiSplitManager implements the ConnectorSplitManager interface. It partitions a Hudi table to read into multiple
+individual chunks (called ConnectorSplit in Presto), so that the data set can be processed in parallel.
+HudiSplitManager also performs partition  pruning if possible.
+
+HudiSplit, which implements ConnectorSplit, describes which files to read, and the file properties (i.e. offset, length,
+location, etc).  For a normal Hudi table, HudiSplitManager generates a split for each FileSlice at certain instant.
+For a Hudi table that contains large base files, HudiSplitManager divides the base file into multiple splits,
+each containing all the log files and part of the base file. This should improve the performance.
+
+For large Hudi tables, generating splits tends to cost lots of time. This can be improved in the following ways. 
+Firstly, increase parallelism. Since each partition is independent of another, processing the partitions in multiple
```

Comment:
There is also an overhead of fetching partitions. Are we going to use metastore to fetch partitions or Hudi API?

Reply:
We are going to use metastore to fetch partitions. This will introduce some RPC, which usually costs less than 100ms.